### DISCUSSION TITLE
don't delete the backend table when you delete a collection

### DISCUSSION THREAD

**Discussion by JetterMcTedder** [directus/directus#11420] [Feature Requests]
just a suggestion but make it an option or a section to not/to delete the backend table that a collection is created on. I can see this bitting someone in the butt at some point in the future....

**Comment by rijkvanzanten** [comment]
What would you expect to happen when you delete a collection instead? 🤔

**Reply by JetterMcTedder** [reply]
delete the collection in the UI but leave the table as is. I could see someone using this and not being aware of this behavior. They put a collection over a table to try it out. They then delete the collection for some reason. They have no idea the table and all the data in the table just got dumped.

The directus_collections table could act as a skin over the table when needed...a switch on the table which is Y/N for delete table when collection is deleted. Always have it set to N unless changed in the options for the collection. That would prevent accidental deleting of the table.

Option on the /admin/settings/data-model/collectionName page. Delete Table or something. Add a column to the directus_collections table and check before you delete the table.

**Reply by benhaynes** [reply]
We could also improve this message to be clear that the database table will also be deleted...

![Screen Shot 2022-02-03 at 6 06 09 PM](https://user-images.githubusercontent.com/522079/152444213-85bbdd93-8534-401e-abb1-ec1c7a101b79.png)

**Reply by JetterMcTedder** [reply]
but what if i just dont want to have a REST interface to my table anymore? There is no option to just remove the collection from the table.

**Reply by rijkvanzanten** [reply]
> but what if i just dont want to have a REST interface to my table anymore?

Right now, Directus' APIs mirror the database 1-1 (that's by design). There's an env-var to exclude certain tables, but registering/deregistering tables like that is not currently something it supports 👍🏻

**Reply by JetterMcTedder** [reply]
might be something to look into this registering/deregistering tables

**Reply by VaZark** [reply]
It could just be marked under an "Archived collections" tab or hidden under some option. This ensures that the UI still reflects the DB and will also allow the admin to delete the table from the db when they choose to "remove permanently" from the archived collections list.

**Reply by benhaynes** [reply]
This makes me wonder if we should just add an "Archive" state for collections. That would allow you to completely disable them within all of Directus, keeps your Directus collection settings available for unarchiving later, and keeps the database table intact. 🤔

Thoughts on that?

**Reply by benhaynes** [reply]
Haha, @VaZark ... I started this message earlier and forgot to hit send, glad we're thinking the same! 🧠↔️🧠

**Reply by JetterMcTedder** [reply]
thanks for following up with this and it sounds like you have a plan!

**Reply by vanling** [reply]
This makes sense to me because we also have the ability to add/connect/link existing tables. So instead of deleting the DB you could also "unlink" it. There might be some issues with this idea tho when thinking about relations..

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
